lib/media/track/videotrack.js

```diff
@@ -104,7 +104,12 @@ class VideoTrack extends MediaTrack {
                 .drawImage(outputFrame, 0, 0, this._outputFrame.width, this._outputFrame.height);
             }
           })
-          .finally(() => captureFrame(process));
+          .finally(() => {
+            const { enabled, muted, readyState } = this.mediaStreamTrack;
+            if (enabled && !muted && readyState === 'live') {
+              captureFrame(process);
+            }
```

Comment:
I think this should be checked before processing the current frame instead of scheduling the next frame. WDYT?